src/main/java/gov/nasa/pds/harvest/mq/ManagerCommandConsumer.java

```diff
@@ -0,0 +1,79 @@
+package gov.nasa.pds.harvest.mq;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import gov.nasa.pds.harvest.cfg.RegistryCfg;
+import gov.nasa.pds.harvest.dao.RegistryManager;
+import gov.nasa.pds.harvest.mq.msg.ManagerMessage;
+import gov.nasa.pds.registry.common.es.service.ProductService;
+
+/**
+ * Consumes manager commands
+ * @author karpenko
+ */
+public class ManagerCommandConsumer
+{
+    protected Logger log;
+    
+    
+    /**
+     * Constructor
+     * @param cfg registry (Elasticsearch) configuration
+     * @throws Exception an exception
+     */
+    public ManagerCommandConsumer(RegistryCfg cfg) throws Exception
+    {
+        log = LogManager.getLogger(this.getClass());
+    }
+
+    
+    /**
+     * Process manager command message
+     * @param msg collection inventory message
+     * @return true if the message has been processed.
+     * false if the message has to be re-queued and reprocessed.
+     */
+    public boolean processMessage(ManagerMessage msg)
+    {
+        if(msg == null || msg.command == null) return true;
+        
+        String cmd = msg.command;
+        log.info("Processing command " + cmd);
+        
+        if("SET_ARCHIVE_STATUS".equals(cmd))
```

Comment:
@tdddblog I was thinking we could have a more general internal command (for the future) like UPDATE_FIELD but this is good to start with. We'll see when there we be more of this type of update.